.github/workflows/root.yml

```
@@ -0,0 +1,108 @@
+name: Nightly root
+on:
+  schedule:
+   - cron: "0 1 * * *"
+  workflow_dispatch: ~
+permissions:
+  contents: read
+jobs:
+  GENERATE_MATRIX:
+    name: Generate Matrix
+    if: github.repository == 'php/php-src' || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.set-matrix.outputs.branches }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Set fetch-depth to 0 to clone the full repository
+          # including all branches. This is required to find
+          # the correct commit hashes.
+          fetch-depth: 0
+      - name: Grab the commit mapping
+        uses: actions/cache@v4
+        with:
+          path: branch-commit-cache.json
+          # The cache key needs to change every time for the
+          # cache to be updated after this job finishes.
+          key: nightly-${{ github.run_id }}-${{ github.run_attempt }}
+          restore-keys: |
+            nightly-
+      - name: Generate Matrix
+        id: set-matrix
+        run: php .github/nightly_matrix.php "${{ github.event_name }}" "${{ github.run_attempt }}" "${{ github.head_ref || github.ref_name }}"
+      - name: Notify Slack
+        if: failure()
+        uses: ./.github/actions/notify-slack
+        with:
+          token: ${{ secrets.ACTION_MONITORING_SLACK }}
+  NIGHTLY_MASTER:
+    needs: GENERATE_MATRIX
```

Specifying a `name:` would make this somewhat more readable, ditto below.
```suggestion
  NIGHTLY_MASTER:
    name: "master"
    needs: GENERATE_MATRIX
```